### PR TITLE
Dump exp: remove, minify defer statements and use log.Fatal instead of panic in test

### PR DESCRIPTION
**Pull request by arxdsilva**
acording to 7e21a21c7663d88a7787270accddb50cc654e088, [this](https://github.com/hajimehoshi/ebiten/blob/7e21a21c7663d88a7787270accddb50cc654e088/graphicscontext.go#L145) could be dropped, also made a few small related changes.

### CODE REVIEW

File: image_test.go [hunk 1]
```diff
@@ -44,7 +45,7 @@ func TestMain(m *testing.M) {
 		return regularTermination
 	}
 	if err := Run(f, 320, 240, 1, "Test"); err != nil && err != regularTermination {
-		panic(err)
+		log.Fatal(err)
```
Comment (hajimehoshi) [inline image_test.go]: Why is log preferred?

File: graphicscontext.go [hunk 2]
```diff
@@ -142,7 +142,6 @@ func (c *graphicsContext) Update(afterFrameUpdate func()) error {
 			if err != nil {
 				return err
 			}
-			defer f.Close()
```
Comment (hajimehoshi) [inline graphicscontext.go]: Why did you remove this?

Comment (arxdsilva) [reply graphicscontext.go]: you're already closing it a few lines below

Comment (hajimehoshi) [reply graphicscontext.go]: `Close` is not called when `Encode` fails. This just ensures `Close` is called.

**Comment by hajimehoshi** [comment]
BTW, dump-exp is just for experiment, and I don't have a plan to maintain this. Do you have any reason you want to add such fix?

File: ebitenutil/loadimage.go [hunk 3]
```diff
@@ -38,9 +38,7 @@ func NewImageFromFile(path string, filter ebiten.Filter) (*ebiten.Image, image.I
 	if err != nil {
 		return nil, nil, err
 	}
-	defer func() {
-		_ = file.Close()
-	}()
```
Comment (hajimehoshi) [inline ebitenutil/loadimage.go]: This change is fine on master, not dump-exp branch.

File: graphicscontext.go (hunk 2 above)
Comment (arxdsilva) [reply graphicscontext.go]: ye, sorry didnt think about this

File: image_test.go (hunk 1 above)
Comment (arxdsilva) [reply image_test.go]: that's something that I prefer, you also get the os.exit(1) + error without all the stack trace from panic

**Comment by arxdsilva** [comment]
no particular reason, these things just caught my eyes in a fast look
